Source/Scene/PostProcess.js

```diff
@@ -0,0 +1,570 @@
+        //>>includeEnd('debug');
+
+        this._fragmentShader = fragmentShader;
+        this._uniformValues = options.uniformValues;
+        this._textureScale = textureScale;
+        this._forcePowerOfTwo = defaultValue(options.forcePowerOfTwo, false);
+        this._sampleMode = defaultValue(options.samplingMode, PostProcessSampleMode.NEAREST);
+        this._pixelFormat = pixelFormat;
+        this._pixelDatatype = defaultValue(options.pixelDatatype, PixelDatatype.UNSIGNED_BYTE);
+        this._clearColor = defaultValue(options.clearColor, Color.BLACK);
+
+        this._uniformMap = undefined;
+        this._command = undefined;
+
+        this._colorTexture = undefined;
+        this._depthTexture = undefined;
+
+        this._actualUniformValues = {};
+        this._dirtyUniforms = [];
+        this._texturesToRelease = [];
+        this._texturesToCreate = [];
+        this._texturePromise = undefined;
+
+        this._passState = new PassState();
+        this._passState.scissorTest = {
+            enabled : true,
+            rectangle : defined(options.scissorRectangle) ? BoundingRectangle.clone(options.scissorRectangle) : new BoundingRectangle()
+        };
+
+        this._ready = true;
+
+        this._name = options.name;
+        if (!defined(this._name)) {
+            this._name = createGuid();
+        }
+
+        // set by PostProcessCollection
+        this._collection = undefined;
+        this._index = undefined;
+
+        /**
+         * Whether or not to execute this post-process when ready.
+         *
+         * @type {Boolean}
+         */
+        this.enabled = true;
+        this._enabled = this.enabled;
+    }
+
+    defineProperties(PostProcess.prototype, {
+        /**
+         * Determines if this post-process is ready to be executed.
+         *
+         * @memberof PostProcess.prototype
+         * @type {Boolean}
+         * @readonly
+         */
+        ready : {
+            get : function() {
+                return this._ready;
+            }
+        },
+        /**
+         * The unique name of this post-process for reference by other processes in a {@link PostProcessComposite}.
+         *
+         * @memberof PostProcess.prototype
+         * @type {String}
+         * @readonly
+         */
+        name : {
+            get : function() {
+                return this._name;
+            }
+        },
+        /**
+         * The fragment shader to use when execute this post-process.
+         * <p>
+         * The shader must contain a sampler uniform declaration for <code>colorTexture</code>, <code>depthTexture</code>,
+         * or both.
+         * </p>
+         * <p>
+         * The shader must contain a <code>vec2</code> varying declaration for <code>v_textureCoordinates</code> for sampling
+         * the texture uniforms.
+         * </p>
+         *
+         * @memberof PostProcess.prototype
+         * @type {String}
+         * @readonly
+         *
+         * @example
```

Comment:
Could probably remove this example given the example in the constructor function.